In [1]:
#### Question 4

In [1]:
import pandas as pd

In [2]:
### Reading the 'json' file with read_json() and storing it in 'data' variable
data=pd.read_json('https://data.nasa.gov/resource/y77d-th95.json')

In [3]:
data.head(3)

name  id nametype recclass      mass  fall                     year  \
0  Aachen   1    Valid       L5      21.0  Fell  1880-01-01T00:00:00.000   
1  Aarhus   2    Valid       H6     720.0  Fell  1951-01-01T00:00:00.000   
2    Abee   6    Valid      EH4  107000.0  Fell  1952-01-01T00:00:00.000   

     reclat    reclong                                        geolocation  \
0  50.77500    6.08333  {'type': 'Point', 'coordinates': [6.08333, 50....   
1  56.18333   10.23333  {'type': 'Point', 'coordinates': [10.23333, 56...   
2  54.21667 -113.00000  {'type': 'Point', 'coordinates': [-113, 54.216...   

   :@computed_region_cbhk_fwbd  :@computed_region_nnqa_25f4  
0                          NaN                          NaN  
1                          NaN                          NaN  
2                          NaN                          NaN

In [4]:
### Dropping columns ':@computed_region_cbhk_fwbd' and	':@computed_region_nnqa_25f4' as they are not essential 
### as per our problem statement
data = data.drop(labels=[':@computed_region_cbhk_fwbd',':@computed_region_nnqa_25f4'],axis=1)

In [5]:
data['year'].isnull().value_counts()

False    999
True       1
Name: year, dtype: int64

In [6]:
###In the pandas library the minimum representable datetime value is jan 1,1583.Hence the years preceding '1583' are filled with 0 and the years of 
### all the remaining records are extracted
data['year']=pd.to_datetime(data['year'],errors='coerce').dt.year.fillna('0').astype('int')

In [7]:
data['geolocation'].isnull().value_counts()

False    988
True      12
Name: geolocation, dtype: int64

In [8]:
### As per the problem statement let us extract the point coordinates from the 'geolocation columns and store it in separate column 'point_coordinates''
data['point_coordinates']=data['geolocation'].str.get('coordinates')

In [9]:
data.head()

name   id nametype     recclass      mass  fall  year    reclat  \
0    Aachen    1    Valid           L5      21.0  Fell  1880  50.77500   
1    Aarhus    2    Valid           H6     720.0  Fell  1951  56.18333   
2      Abee    6    Valid          EH4  107000.0  Fell  1952  54.21667   
3  Acapulco   10    Valid  Acapulcoite    1914.0  Fell  1976  16.88333   
4   Achiras  370    Valid           L6     780.0  Fell  1902 -33.16667   

     reclong                                        geolocation  \
0    6.08333  {'type': 'Point', 'coordinates': [6.08333, 50....   
1   10.23333  {'type': 'Point', 'coordinates': [10.23333, 56...   
2 -113.00000  {'type': 'Point', 'coordinates': [-113, 54.216...   
3  -99.90000  {'type': 'Point', 'coordinates': [-99.9, 16.88...   
4  -64.95000  {'type': 'Point', 'coordinates': [-64.95, -33....   

      point_coordinates  
0     [6.08333, 50.775]  
1  [10.23333, 56.18333]  
2      [-113, 54.21667]  
3     [-99.9, 16.88333]  
4   [-64.95, -33.16667]

In [10]:
### As per the problem statement , the 'geolocation' column is no longer required, hence dropping it
data = data.drop(labels=['geolocation'],axis=1)

In [11]:
data.head(3)

name  id nametype recclass      mass  fall  year    reclat    reclong  \
0  Aachen   1    Valid       L5      21.0  Fell  1880  50.77500    6.08333   
1  Aarhus   2    Valid       H6     720.0  Fell  1951  56.18333   10.23333   
2    Abee   6    Valid      EH4  107000.0  Fell  1952  54.21667 -113.00000   

      point_coordinates  
0     [6.08333, 50.775]  
1  [10.23333, 56.18333]  
2      [-113, 54.21667]

In [12]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   name               1000 non-null   object 
 1   id                 1000 non-null   int64  
 2   nametype           1000 non-null   object 
 3   recclass           1000 non-null   object 
 4   mass               972 non-null    float64
 5   fall               1000 non-null   object 
 6   year               1000 non-null   int64  
 7   reclat             988 non-null    float64
 8   reclong            988 non-null    float64
 9   point_coordinates  988 non-null    object 
dtypes: float64(3), int64(2), object(5)
memory usage: 78.2+ KB


In [13]:
### Converting the dataframe to the csv file format
data.to_csv('converted_to_csv.csv',index=False)

##### Question 7

In [ ]:
● Get all the Earth meteorites that fell before the year 2000

In [24]:
data[data['year']<2000][['name']]

name
0       Aachen
1       Aarhus
2         Abee
3     Acapulco
4      Achiras
..         ...
994   Timochin
995   Tirupati
997      Tjabe
998   Tjerebon
999  Tomakovka

[930 rows x 1 columns]

In [25]:
a=data[data['year']<2000][['name']]
a['name'].unique()

array(['Aachen', 'Aarhus', 'Abee', 'Acapulco', 'Achiras', 'Adhi Kot',
       'Adzhi-Bogdo (stone)', 'Agen', 'Aguada', 'Aguila Blanca',
       'Aioun el Atrouss', 'Aïr', 'Aire-sur-la-Lys', 'Akaba', 'Akbarpur',
       'Akwanga', 'Akyumak', 'Al Rais', 'Alais', 'Albareto', 'Alberta',
       'Aldsworth', 'Aleppo', 'Alessandria', 'Alexandrovsky',
       'Alfianello', 'Allegan', 'Allende', "Alta'ameem", 'Ambapur Nagla',
       'Andhara', 'Andover', 'Andreevka', 'Andura',
       'Northwest Africa 5815', 'Angers', 'Angra dos Reis (stone)',
       'Ankober', 'Anlong', 'Aomori', 'Appley Bridge', 'Apt',
       'Arbol Solo', 'Archie', 'Arroyo Aguiar', 'Asco', 'Ashdon',
       'Assisi', 'Atarra', 'Atemajac', 'Athens', 'Atoka', 'Aubres',
       'Aumale', 'Aumieres', 'Ausson', 'Avanhandava', 'Avce', 'Avilez',
       'Awere', 'Aztec', 'Bachmut', 'Bahjoi', 'Bald Mountain', 'Baldwyn',
       'Bali', 'Ban Rong Du', 'Bandong', 'Bansur', 'Banswal', 'Banten',
       'Barbotan', 'Barcelona (stone)', 'Barea', 

##### There are a total of 930 Earth meteorites that fell before the year 2000

In [ ]:
● Get all the earth meteorites co-ordinates who fell before the year 1970

In [30]:
b=data[data['year']<1970][['point_coordinates']]
b['point_coordinates']

0          [6.08333, 50.775]
1       [10.23333, 56.18333]
2           [-113, 54.21667]
4        [-64.95, -33.16667]
5               [71.8, 32.1]
               ...          
994             [35.2, 54.5]
995     [79.41667, 13.63333]
997    [111.53333, -7.08333]
998    [106.58333, -6.66667]
999        [34.76667, 47.85]
Name: point_coordinates, Length: 781, dtype: object

#### There are a total of 781 earth meteorites that had fell before the year 1970

In [ ]:
● Assuming that the mass of the earth meteorites was in kg, get all those whose mass was more
than 10000kg

In [33]:
data[data['mass']>10000][['name']]

name
2          Abee
7          Agen
11          Aïr
16      Akyumak
27   Alfianello
..          ...
991   Tieschitz
992      Tilden
994    Timochin
997       Tjabe
998    Tjerebon

[243 rows x 1 columns]

In [36]:
c=data[data['mass']>10000][['name']]
c['name'].unique()

array(['Abee', 'Agen', 'Aïr', 'Akyumak', 'Alfianello', 'Allegan',
       'Allende', 'Andura', 'Appley Bridge', 'Aumale', 'Ausson',
       'Bachmut', 'Bahjoi', 'Ban Rong Du', 'Bandong', 'Bansur', 'Barwell',
       'Bassikounou', 'Baszkówka', 'Bath', 'Bath Furnace', 'Beardsley',
       'Beaver Creek', 'Benguerir', "Beni M'hira", 'Bensour', 'Béréba',
       'Bilanga', 'Bjurböle', 'Boguslavka', 'Braunau', 'Bruderheim',
       'Bur-Gheluai', 'Bursa', 'Butsura', 'Buzzard Coulee',
       'Cabezo de Mayo', 'Cabin Creek', 'Campos Sales', 'Cangas de Onis',
       'Castrovillari', 'Chajari', 'Chantonnay', 'Charsonville',
       'Charwallas', 'Château-Renard', 'Chelyabinsk', 'Chergach ',
       'Colby (Wisconsin)', 'Conquista', 'Cumberland Falls', 'Dahmani',
       'Demina', 'Dergaon', 'Desuri', 'Devgaon', 'Dhajala', 'Dhurmsala',
       'Djati-Pengilon', 'Dong Ujimqin Qi', 'Dresden (Ontario)', 'Durala',
       'El Paso de Aguila', 'Elbogen', 'Elenovka', 'Ensisheim', 'Ergheo',
       'Estherville',

#### There are a total of 243 number of earth meteorites that have mass more than 10000 kgs